Formula/zim.rb

```diff
@@ -29,19 +29,25 @@ class Zim < Formula
   end
 
   def install
-    python_version = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
-    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{python_version}/site-packages"
+    python3 = which("python3")
+    ENV.prepend_create_path "PYTHONPATH", libexec/Language::Python.site_packages("python3")
     resource("pyxdg").stage do
-      system Formula["python@3.9"].opt_bin/"python3", *Language::Python.setup_install_args(libexec/"vendor")
+      system python3, *Language::Python.setup_install_args(libexec/"vendor")
     end
     ENV["XDG_DATA_DIRS"] = libexec/"share"
-    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{python_version}/site-packages"
-    system Formula["python@3.9"].opt_bin/"python3", "./setup.py", "install", "--prefix=#{libexec}"
+    system python3, "./setup.py", "install", "--prefix=#{libexec}"
     bin.install Dir[libexec/"bin/*"]
     bin.env_script_all_files libexec/"bin",
       PYTHONPATH:    ENV["PYTHONPATH"],
-      XDG_DATA_DIRS: ["#{HOMEBREW_PREFIX}/share", libexec/"share"].join(":")
+      XDG_DATA_DIRS: [HOMEBREW_PREFIX/"share", libexec/"share"].join(":")
     pkgshare.install "zim"
+
+    # Make the bottles uniform
+    inreplace [
+      libexec/Language::Python.site_packages("python3")/"zim/config/basedirs.py",
+      libexec/"vendor"/Language::Python.site_packages("python3")/"xdg/BaseDirectory.py",
+      pkgshare/"zim/config/basedirs.py",
+    ], "/usr/local", HOMEBREW_PREFIX
```

Comment:
Just a note: one thing I don't like is that creating `:all` bottles is nice for our users,
but it needs yet another inreplace.

We try to avoid this if possible, because:
- it's fragile
- we are modifying upstream's code, which might create subtle bugs

It's not specific to this formula but a more general thought.

Reply:
I'm just `inreplace`-ing lines that are already touched by `keg_relocate` for `/usr/local` prefixes. We should probably really make `keg_relocate` a bit more selective about what it changes, especially with `/usr/local`, but that's a larger project.

As for modifying upstream's code: this only changes the fallback to `XDG_DATA_DIRS`, which should not be reached given that we set that variable in an env script. If anything, this `inreplace` makes sure the fallback is more consistent with our choice of `XDG_DATA_DIRS`.